drivers/dma/dma_silabs_ldma.c

```diff
@@ -0,0 +1,564 @@
+}
+
+static int dma_silabs_get_status(const struct device *dev, uint32_t channel,
+				 struct dma_status *status)
+{
+	const struct dma_silabs_data *data = dev->data;
+
+	if (channel > data->dma_ctx.dma_channels) {
+		return -EINVAL;
+	}
+
+	if (!atomic_test_bit(data->dma_ctx.atomic, channel)) {
+		return -EINVAL;
+	}
+
+	status->busy = data->dma_chan_table[channel].busy;
+	status->dir = data->dma_chan_table[channel].dir;
+
+	return 0;
+}
+
+static int dma_silabs_init(const struct device *dev)
+{
+	struct dma_silabs_config *config = (struct dma_silabs_config *)dev->config;
+	struct dma_silabs_data *data = (struct dma_silabs_data *)dev->data;
+	struct dma_silabs_channel *dma_channel;
+	LDMA_Init_t dmaInit = LDMA_INIT_DEFAULT;
+
+	if (!device_is_ready(config->clock_dev)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	for (int i = 0; i < data->dma_ctx.dma_channels; i++) {
+		dma_channel = &data->dma_chan_table[i];
+		dma_channel->id = i;
+		dma_channel->busy = false;
+		dma_channel->dir = MEMORY_TO_MEMORY;
+		dma_channel->complete_callback_en = 1;
+		dma_channel->cb = NULL;
+		dma_channel->user_data = NULL;
+		memset(&dma_channel->xfer_config, 0, sizeof(dma_channel->xfer_config));
+	}
+
+	/*  0x7 indicate that the 8 channels have round robin priority. */
+	dmaInit.ldmaInitCtrlNumFixed = 0x7;
+	dmaInit.ldmaInitIrqPriority = DMA_IRQ_PRIORITY;
+
+	LDMA_Init(&dmaInit);
+
+	return 0;
+}
+
+static const struct dma_driver_api dma_funcs = {.config = dma_silabs_configure,
+						.start = dma_silabs_start,
+						.stop = dma_silabs_stop,
+						.get_status = dma_silabs_get_status};
+
+#define SILABS_DMA_IRQ_CONNECT(n, inst)                                                            \
+	IRQ_CONNECT(DT_INST_IRQ_BY_IDX(inst, n, irq), DT_INST_IRQ_BY_IDX(inst, n, priority),       \
+		    dma_silabs_irq_handler, DEVICE_DT_INST_GET(inst), 0);                          \
+	irq_enable(DT_INST_IRQ_BY_IDX(inst, n, irq));
+
+#define CONFIGURE_ALL_IRQS(inst, n) LISTIFY(n, SILABS_DMA_IRQ_CONNECT, (), inst)
+
+#define DMA_SILABS_INIT(inst)                                                                      \
+                                                                                                   \
+	static void silabs_dma_irq_configure_##inst(const struct device *dev)                      \
+	{                                                                                          \
+		ARG_UNUSED(dev);                                                                   \
+		CONFIGURE_ALL_IRQS(inst, DT_NUM_IRQS(DT_DRV_INST(inst)));                          \
+	};                                                                                         \
+                                                                                                   \
+	const struct dma_silabs_config dma_silabs_config_##inst = {                                \
+		.config_irq = silabs_dma_irq_configure_##inst,                                     \
```

Comment:
`config_irq()` is never called, the driver seems to be relying on a mix of the interrupt config happening inside the HAL and the fact that IRQ_CONNECT() uses linker magic, so the ISR gets registered in the vector table by the linker, and is enabled and configured by the HAL.

This combination means that the IRQ priority configured in DT isn't respected by the driver. IMO `config_irq()` should be called after `LDMA_Init()` to rewrite priorities and make the DT configuration stick.